server/cmd/dcrdex/config.go

```diff
@@ -0,0 +1,547 @@
+	// options.
+	if configFileError != nil {
+		fmt.Printf("%v\n", configFileError)
+		return loadConfigError(configFileError)
+	}
+
+	// Append the network type to the data directory so it is "namespaced" per
+	// network.  In addition to the block database, there are other pieces of
+	// data that are saved to disk such as address manager state. All data is
+	// specific to a network, so namespacing the data directory means each
+	// individual piece of serialized data does not have to worry about changing
+	// names per network and such.
+	//
+	// Make list of old versions of testnet directories here since the network
+	// specific DataDir will be used after this.
+	cfg.DataDir = cleanAndExpandPath(cfg.DataDir)
+	cfg.DataDir = filepath.Join(cfg.DataDir, network.String())
+	// Create the data folder if it does not exist.
+	err = os.MkdirAll(cfg.DataDir, 0700)
+	if err != nil {
+		return nil, err
+	}
+
+	logRotator = nil
+	// Append the network type to the log directory so it is "namespaced"
+	// per network in the same fashion as the data directory.
+	cfg.LogDir = cleanAndExpandPath(cfg.LogDir)
+	cfg.LogDir = filepath.Join(cfg.LogDir, network.String())
+
+	// Ensure that the RPC key and certificate are absolute paths.
+	certDir, certFile := filepath.Split(cfg.RPCCert)
+	if !filepath.IsAbs(certDir) {
+		cfg.RPCCert = filepath.Join(cfg.AppDataDir, certFile)
+	}
+	keyDir, keyFile := filepath.Split(cfg.RPCKey)
+	if !filepath.IsAbs(keyDir) {
+		cfg.RPCKey = filepath.Join(cfg.AppDataDir, keyFile)
+	}
+
+	// Validate each RPC listen host:port.
+	var RPCListen []string
+	if len(cfg.RPCListen) == 0 {
+		RPCListen = []string{defaultRPCHost + ":" + defaultRPCPort}
+	}
+	for i := range cfg.RPCListen {
+		listen, err := normalizeNetworkAddress(cfg.RPCListen[i], defaultRPCHost, defaultRPCPort)
+		if err != nil {
+			return loadConfigError(err)
+		}
+		RPCListen = append(RPCListen, listen)
+	}
+
+	// Initialize log rotation. After log rotation has been initialized, the
+	// logger variables may be used. This creates the LogDir if needed.
+	if cfg.MaxLogZips < 0 {
+		cfg.MaxLogZips = 0
+	}
+	initLogRotator(filepath.Join(cfg.LogDir, defaultLogFilename), cfg.MaxLogZips)
+
+	log.Infof("Log folder:  %s", cfg.LogDir)
+	log.Infof("Config file: %s", configFile)
+
+	// Special show command to list supported subsystems and exit.
+	if cfg.DebugLevel == "show" {
+		fmt.Println("Supported subsystems", supportedSubsystems())
+		os.Exit(0)
+	}
+
+	// Ensure HTTP profiler is mounted with a valid path prefix.
+	// if cfg.HTTPProfile && (cfg.HTTPProfPath == "/" || len(defaultHTTPProfPath) == 0) {
+	// 	return loadConfigError(fmt.Errorf("httpprofprefix must not be \"\" or \"/\""))
+	// }
+
+	// Parse, validate, and set debug log level(s).
+	if err := parseAndSetDebugLevels(cfg.DebugLevel); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		parser.WriteHelp(os.Stderr)
+		return loadConfigError(err)
+	}
+
+	dbHost, dbPortStr, err := net.SplitHostPort(cfg.PGHost)
```

Comment:
This prevents unix sockets like /var/run/postgresql.  Needs to work more like dcrdata for the PGHost.